src/com/xilinx/rapidwright/design/tools/ArrayBuilder.java

```diff
@@ -766,16 +951,26 @@ public static void main(String[] args) {
             array.setAutoIOBuffers(false);
         }
 
-        Net gndNet = array.getNet(Net.GND_NET);
-        gndNet.unroute();
-        Net vccNet = array.getNet(Net.VCC_NET);
-        vccNet.unroute();
+        if (ab.unrouteStaticNets()) {
+            Net gndNet = array.getNet(Net.GND_NET);
+            if (gndNet != null) {
+                gndNet.unroute();
+            }
+            Net vccNet = array.getNet(Net.VCC_NET);
+            if (vccNet != null) {
+                vccNet.unroute();
+            }
+        }
         array.getNetlist().consolidateAllToWorkLibrary();
+        array.flattenDesign();
 
-        if (ab.getOutOfContext()) {
+        if (ab.isOutOfContext()) {
             // Automatically find bounding PBlock based on used Slices, DSPs, and BRAMs
             Set<Site> usedSites = new HashSet<>();
             for (SiteInst siteInst : array.getSiteInsts()) {
+                if (siteInst.getName().contains("STATIC_SOURCE_SLICE")) {
```

Comment:
Where is `STATIC_SOURCE_SLICE` defined?  There is a `SiteInst.STATIC_SOURCE` string that is `STATIC_SOURCE`.